src/pages/user-descriptor/weight-area-chart.jsx

```diff
@@ -1,50 +1,71 @@
 import React from "react";
 import ReactEcharts from "echarts-for-react"; // or var ReactEcharts = require('echarts-for-react');
-
-// Expected Object
-function Datapoint(uunit, uvalue, ulat, ulon, utime, ugender) {
-  this.uunit = uunit;
-  this.uvalue = uvalue;
-  this.ulat = ulat;
-  this.ulon = ulon;
-  this.utime = utime;
-  this.ugender = ugender;
+import { gql } from 'apollo-boost';
+import { useQuery } from '@apollo/react-hooks';
+
+//GraphQL ----------------------------------------------------------------
+const GET_POUNDS = gql`
+      {
+        getPaginatedDescriptors(unit: "lb", start: 0, count: 500)
+        {value}
+      }
+      `;
+
+function POUNDS() {
+  const { loading, error, data } = useQuery(GET_POUNDS);
+
+  if (loading) return 'Loading...';
+  if (error) return `Error! ${error.message}`;
```

Comment:
Be careful. The data that you return is an array if successful, but when unsuccessful, you return a string. In your render code, you have the assumption that this function returns an array, but this isn't always the case. When interacting with a real network, data may take seconds or completely fail to come through so you need to show that the data is loading to the user, or if there is an error. https://www.apollographql.com/docs/react/get-started/#request-data